src/Endpoints/class-graphql-metadata.php

```diff
@@ -0,0 +1,266 @@
+<?php
+/**
+ * GraphQL support class.
+ * Note: this class will only work if the WPGraphQL plugin is installed.
+ *
+ * @package Parsely
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Endpoints;
+
+use WP_Post;
+
+/**
+ * Injects Parse.ly Metadata to the GraphQL outputs
+ *
+ * @since 3.2.0
+ */
+class GraphQL_Metadata extends Metadata_Endpoint {
+	private const GRAPHQL_VERSION          = '1.0.0';
+	private const GRAPHQL_CONTAINER_TYPE   = 'ParselyMetaContainer';
+	private const GRAPHQL_AUTHOR_TYPE      = 'ParselyAuthor';
+	private const GRAPHQL_MAIN_ENTITY_TYPE = 'ParselyMainEntityOfPage';
+	private const GRAPHQL_IMAGE_TYPE       = 'ParselyImage';
+	private const GRAPHQL_PUBLISHER_TYPE   = 'ParselyPublisher';
+	private const GRAPHQL_META_TYPE        = 'ParselyMeta';
+
+	/**
+	 * Register fields in WPGraphQL plugin
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		if ( class_exists( 'WPGraphQL' ) ) {
+			/**
+			 * Filter whether GraphQL support is enabled or not.
+			 *
+			 * @since 3.2.0
+			 *
+			 * @param bool $enabled True if enabled, false if not.
+			 */
+			if ( apply_filters( 'wp_parsely_enable_graphql_support', true ) && $this->parsely->api_key_is_set() ) {
+				add_action( 'graphql_register_types', array( $this, 'register_meta' ) );
+			}
+		}
+	}
+
+	/**
+	 * Registers the meta field on the appropriate resource types in the REST API.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	public function register_meta(): void {
+		$this->register_object_types();
+		$this->register_fields();
+	}
+
+	/**
+	 * Registers the new custom types for Parse.ly Metadata into the GraphQL instance.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	private function register_object_types(): void {
```

Comment:
I would suggest breaking this method down into smaller methods, one for each type of object graph, and then have this method call those new methods. It's more granular, can be better documented (now, and with any future changes that only affect one object type), avoids an 84-line function call with 6 different sets of argument arrays, and with reflection can be integration tested for each individual registration.

Reply:
Given that we're pulled out all types but one, I think it's fine to leave the function as is.